### DISCUSSION TITLE
Is it possible to hide specific parametrize markers ids?

### DISCUSSION THREAD

**Discussion by panjacek** [pytest-dev/pytest#8334] [Q&A] [answered]
I am writing optional finalizer which has to be executed late and I figure out that I can decide if its run via mark.parametrize with indirect, but I don't want to see my bool as id of test case as this will look dirty.

```python
@pytest.fixture(autouse=True)
my_logcatcher():
  catch_logs()
  if something is True:
     parse_logs()

@pytest.mark.parametrize("my_logcatcher", [True], indirect=True, scope="function")
test_something_X():
  pass

test_something_Y():
  pass
```

I would like to skip addition of that True to test case name.

On the other side maybe there is a cleaner way to do what I want.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
Just use a different mark and check for the mark in the fixture

**Reply by panjacek** [reply]
great thanks for quick idea, works as intended. Posting what I've used for future reference.

```python
@pytest.mark.my_marker
test_something_X():
  pass


@pytest.fixture(autouse=True)
my_logcatcher():
  catch_logs()
  if request.node.get_closest_marker("my_marker"):
     parse_logs()
```
